src/realtime/realtime-hapi-plugin.ts

```diff
@@ -146,45 +157,54 @@ export class RealtimeHapiPlugin {
     } as PersistedEvent<any>;
   }
 
-  private async onRequest(teamId: number, since?: number) {
-      let observable = Observable.concat(
-        Observable.of(this.pingEvent()),
-        this.persistedEvents.filter(event => event.teamId === teamId),
-      );
-      if (since) {
-        const existing = await this.eventBus.getEvents(teamId, since);
-        if (existing.length > 0) {
-          existing.shift(); // getEvents is '>= since', but here we want '> since'
-        }
-        observable = Observable.concat(Observable.from(existing), observable);
+  private async getStream(request: Hapi.Request) {
+    const {teamId, projectId, deploymentId} = request.params;
+    const _teamId = parseInt(teamId, 10);
+    const since = parseInt(request.headers['last-event-id'], 10);
+    let predicate = (event: PersistedEvent<any>) => event.teamId === _teamId;
+    if (projectId && deploymentId) {
+      const _projectId = parseInt(projectId, 10);
+      const _deploymentId = parseInt(deploymentId, 10);
+      predicate = deploymentEventFilter(_teamId, _projectId, _deploymentId);
+    }
+    let observable = Observable.concat(
+      Observable.of(this.pingEvent()),
+      this.persistedEvents.filter(predicate),
+    );
+    if (since && !isNaN(since)) {
+      const existing = await this.eventBus.getEvents(_teamId, since);
```

Comment:
Will this pass on missed events that are not related to this deployment if `projectId` and `deploymentId` exist?